scripts/compiling.py

```diff
@@ -0,0 +1,322 @@
+import lxml.etree as ET
+import sys
+import os
+
+# shorthand for yaml tabulation
+def getTabs(n):
+	str=""
+	for x in range(n):
+		str+="  "
+	return str
+
+def processSubProperty(prop):
```

Comment:
Why this function is not returning anything? For example a `True/False` to check the functionality

Reply:
Wouldn't this better be managed with exceptions? Throw an exception the main method would catch and take that possible exception as a way to manage errors? Did you mean manage those possible exceptions locally (inside the method itself) and then return a simple boolean?

Reply:
Added a (very) primitive exception handling structure in https://github.com/erlerobot/HRIM/pull/16/commits/0139f99ff04bb68c9e6637e794434ad96dbf53d7.

By primitive I mean they're try: blocks encompassing full functions and except: blocks taking ALL exceptions, printing a method-specific message and raising the exception to it's top level, providing some information on where the exception happened but not what it was.

Examples of (forced) exceptions:
```
Error while processing subproperty message: Time
Error while processing subproperty message: Header
Error while compiling generic package
An error occurred during command execution
```
```
Error while processing subproperty message: SpecsForceAxis
Error while processing topic message: SpecsForce
Error while processing module: force
An error occurred during command execution
```
```
Error parsing dataType mapping file
Error while compiling generic package
An error occurred during command execution
```